src/Data/Constraint/Extras.hs

```diff
@@ -28,37 +29,27 @@ import Data.Constraint.Forall
 -- applied didn't quite cut it when I tried). Some symbolic type-level application could do the trick, but I didn't
 -- want to go quite that far at the time of writing.
 
-class ArgDict (c :: k -> Constraint) (f :: k -> *) where
-  type ConstraintsFor f (c :: k -> Constraint) :: Constraint
-  argDict :: ConstraintsFor f c => f a -> Dict (c a)
+class Has c f where
+  has :: forall a r. f a -> (c a => r) -> r
+  has x r | Dict <- argDict @c x = r
+  argDict :: forall a. f a -> Dict (c a)
+  argDict x = has @c x Dict
```

Comment:
Why is this a method?

Reply:
Made it so that the TH change was more trivial.

Reply:
also, it helps people who have existing handwritten instances of the original ArgDict -- they can keep the implementation the same